### ISSUE TITLE
Add ability to specify which Python versions to run with from the command line.

### ISSUE THREAD

**Issue by theacodes** [theacodes/nox#95]
Context: https://github.com/theacodes/nox/pull/94/files/ad56baafab381ef967f504b1c2d9dc30d61bc96e#r205984653

**Comment by stsewd** [comment]
What about implicit specifying the python version on the session name? Something like:

```python

@nox.session(python=['3.5', '3.6'])
def test():
    ...
```

```bash
$ nox -s test-3.7
nox> Warning: running session with custom interpreter
...
```

**Comment by theacodes** [comment]
I think that's a separate use case, but a useful one. Wanna file a separate bug for it?

**Comment by stsewd** [comment]
I opened https://github.com/theacodes/nox/issues/158

**Comment by jarus** [comment]
This would be a super helpful enhancement and would make nox even more useful, especially in situations where the developer would like to run a session(s) with one or multiple selected python version(s).

Do you have already a favourite solution for this problem which you would like to see in nox? 

As a short term solution, I think it would be sufficient to add the python version in a normalized form (`3.7` => `py37`) to the keyword list beside the `SessionRunner.signature` so that `nox -k py37` only select sessions with the python 3.7 interpreter. Is that something you would welcome? If yes, I can create a pull request.

In general, I like nox's and pytest's intention to reuse an already known and established syntax like Python's logical operators for selecting sessions or tests but the current simple implementations with Python `eval` function also come with some disadvantages. For example it blocks the usage of keywords that don't follow the Python variable name rule like `3.7`, `test-3.7` or `from`. Therefore it is maybe a good idea to think about replacing `eval` with an own matcher with a more flexible syntax.

**Comment by theacodes** [comment]
@jarus, that sounds great - do you want to create a separate issue for that so we can track it? I'm more than happy to see a PR for that.

**Comment by theacodes** [comment]
This issue is actually a great first issue. It involves:

1. Adding a new command-line parameter `--python` (or similar)
2. Updating `manifest`'s filtering to filter based on the `--python` parameter.

**Comment by theacodes** [comment]
It should work as an additional filter to the manifest. `--session` filters
the manifest by name, `--keyword` filters the manifest through pattern
matching, and `--python` should filter the manifest by python version.

**Comment by maciej-lech** [comment]
As @vikrambhat11 has not been prepared a PR for this one, I'd like to propose my implementation as it would be a more convenient way for filtering by python interpreter versions comparing to using keywords.

**Comment by pradyunsg** [comment]
This can be closed, given #304 is merged?

**Comment by theacodes** [comment]
Yup, thanks.
